ArmoniK.Extension.CSharp.DllWorker/LibraryLoader.cs

```diff
@@ -0,0 +1,404 @@
+                                        CancellationToken cancellationToken)
+  {
+    try
+    {
+      Logger.LogInformation("Starting to LoadLibrary");
+      Logger.LogInformation($"Nb of current loaded assemblies: {assemblyLoadContexts_.Count}");
+
+      var taskLibrary = taskHandler.TaskOptions.GetServiceLibrary();
+
+      //Get the data about the dynamic library
+      var dynamicLibrary = taskHandler.TaskOptions.GetTaskLibraryDefinition(taskLibrary);
+      var filename       = $"{dynamicLibrary}.zip";
+
+      var filePath = @"/tmp/zip";
+
+      var destinationPath = @"/tmp/assemblies";
+
+      var pathToDllFile = dynamicLibrary.PathToFile;
+
+      var dllFileName = dynamicLibrary.DllFileName;
+
+      Logger.LogInformation($"Starting Dynamic loading - TaskLibrary: {taskLibrary}, FileName: {filename}, FilePath: {filePath}, DestinationToUnZip:{destinationPath}, PathToDllFile:{pathToDllFile}, DllFileName: {dllFileName}, Namespace: {dynamicLibrary.Namespace}, Service: {dynamicLibrary.Service}");
+
+      //if the context is already loaded
+      if (assemblyLoadContexts_.ContainsKey(dynamicLibrary.ToString()))
+      {
+        return dynamicLibrary.ToString();
+      }
+
+      var loadContext = new AssemblyLoadContext(dynamicLibrary.ToString());
+
+      var dllExists = taskHandler.DataDependencies.TryGetValue(dynamicLibrary.LibraryBlobId,
+                                                               out var libraryBytes);
+      if (!dllExists || libraryBytes == null)
+      {
+        throw new WorkerApiException("No library found on data dependencies.");
+      }
+
+      try
+      {
+        Directory.CreateDirectory(filePath);
+
+        // Create the full path to the zip file
+        var zipFilePath = Path.Combine(filePath,
+                                       filename);
+
+        await File.WriteAllBytesAsync(zipFilePath,
+                                      libraryBytes,
+                                      cancellationToken);
+        /*
+        await using var fs = new FileStream(Path.Join(filePath,
+                                                      filename),
+                                            FileMode.Create,
+                                            FileAccess.ReadWrite,
+                                            FileShare.Read,
+                                            libraryBytes.Length);
+
+        fs.Seek(0,
+                SeekOrigin.Begin);
+        await fs.WriteAsync(libraryBytes,
+                            0,
+                            libraryBytes.Length,
+                            cancellationToken);
+        */
+      }
+      catch (Exception ex)
+      {
+        Logger.LogError("W");
+        throw new WorkerApiException(ex);
+      }
+
+      Logger.LogInformation("Extracting from archive {localZip}",
+                            Path.Join(filePath,
+                                      filename));
+
+      var extractedPath = ExtractArchive(filename,
```

Comment:
Do you have to create a temporary zipfile ? Can't you input the bytes directly into the function ?

Reply:
I did not find a way to do it straight from the bytes. We talked on daily as something that could be enhanced later, but to keep the attention.